### DISCUSSION TITLE
Unable to query using $some/$none

### DISCUSSION THREAD

**Discussion by F5AlexB** [mikro-orm/mikro-orm#5995] [General]
### Describe the bug

When using $some or $none, a TypeError is received from the code if this is done at the root node of the query.
If instead added as part of an AND statement, a query is generated, but the query is invalid (misnamed fields which are attributed to the wrong table in the query).

```
const standalone2Repo = orm.em.getRepository(StandaloneEntityTwo);

  await standalone2Repo.find({
    $and: [
      {
        someBooleanFlag: true,
      },
      {
        name: { $like: '%asdf%' }
      },
      {
        listOfMiddleEntitites: {
          $none: {
            standalone1: {
              id: 1,
            },
            arbitraryData: { $ne: null },
          }
        }
      }
    ],
  });
```

**Result/Output:**
```
InvalidFieldNameException: select `s0`.* from `standalone_entity_two` as `s0` where `s0`.`some_boolean_flag` = true and `s0`.`name` like '%asdf%' and (`s0`.`standalone1`, `s0`.`standalone2`) not in (select `s0`.`standalone1`, `s0`.`standalone2` from `standalone_entity_two` as `s0` inner join `an_entity_in_the_middle` as `a1` on `s0`.`field_id` = `a1`.`standalone2_field_id` where `a1`.`standalone1_id` = 1 
and `a1`.`arbitrary_data` is not null) - SQLITE_ERROR: no such column: s0.standalone1
````

### Reproduction

https://github.com/F5AlexB/mikro-orm_reproduction_nested-relations

### What driver are you using?

@mikro-orm/sqlite - Though fairly certain this impacts other drivers as we first saw this issue using postgres.

### MikroORM version

6.3.7

### Node.js version

20.12.2

### Operating system

Windows 10

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> no such column: s0.standalone1

This is already fixed in master, try latest dev version (I might actually ship this to stable during the morning).

Your third test case is still failing, but I don't see much need to support `$not->$some` when that is literally what the `$none` operator is for. I will take a closer look and at least add a validation if I don't see an easy way out, as we surely don't want to fail with TypeError, that's granted.

**Reply by B4nan** [reply]
Actually, the third test case was failing because of the `$not: { arbitraryData: null }` query, not the `$not->$some` combination, that one actually works fine apparently. You should be using `arbitraryData: { $ne: null }` like in the first two test cases.

**Reply by F5AlexB** [reply]
Thank you for the prompt replies.

Apologies for raising this - was not aware that there was already a fix in for it pending release.

Regarding the $not: { arbitraryData: null } approach - is this causing a TypeError a problem? Or as expected?

**Reply by B4nan** [reply]
It's definitely not something you should be doing, the correct way is the `$ne: null` operator which represents `is not null`, what you tried should end up as `not (sth is null)` instead of `sth is not null` (and I don't understand why would you want that).
